runtime/lua/builtin_util.lua

```diff
@@ -0,0 +1,90 @@
+
+local util = {}
+
+vim = vim or {}
+
+util.echom = function(message)
+  vim.api.nvim_command('echom "' .. tostring(message) .. '"')
+end
+
+util.split = function(s, sep, nMax, bRegexp)
+  assert(sep ~= '')
+  assert(nMax == nil or nMax >= 1)
+
+  local aRecord = {}
+
+  if s:len() > 0 then
+    local bPlain = not bRegexp
+    nMax = nMax or -1
+
+    local nField, nStart = 1, 1
+    local nFirst, nLast = s:find(sep, nStart, bPlain)
+    while nFirst and nMax ~= 0 do
+      aRecord[nField] = s:sub(nStart, nFirst - 1)
+      nField = nField + 1
+      nStart = nLast + 1
+      nFirst, nLast = s:find(sep, nStart, bPlain)
+      nMax = nMax - 1
+    end
+
+    aRecord[nField] = s:sub(nStart)
+  end
+
+  return aRecord
+end
+
+util.tostring = function(obj)
+  local stringified = ''
+  if type(obj) == 'table' then
+    stringified = stringified .. '{'
+    for k, v in pairs(obj) do
+      stringified = stringified .. tostring(k) .. '=' .. tostring(v) .. ','
+    end
+    stringified = stringified .. '}'
+  else
+    stringified = tostring(obj)
+  end
+
+  return stringified
+end
+
+
+util.trim = function(s)
+  return s:gsub("^%s+", ""):gsub("%s+$", "")
+end
+
+
+util.handle_uri = function(uri)
+  local file_prefix = 'file://'
+  if string.sub(uri, 1, #file_prefix) == file_prefix then
+    return string.sub(uri, #file_prefix + 1, #uri)
+  end
+
+  return uri
+end
+
+
+-- Determine whether a Lua table can be treated as an array.
+-- Returns:
+--  true    A non-empty array
+--  false   A non-empty table
+--  nil     An empty table
+util.is_array = function(table)
+  local count = 0
+
+  for k, v in pairs(table) do
+    if type(k) == "number" then
+      count = count + 1
+    else
+      return false
+    end
+  end
+
+  if count > 0 then
+    return true
```

Comment:
What about simply:

```lua
return count > 0
```

But even better, in the lua docs we find:

```
Unless a __len metamethod is given, the length of a table t is only defined if the table is a sequence, that is, the set of its positive numeric keys is equal to {1..n} for some integer n. In that case, n is its length. Note that a table like

  {10, 20, nil, 40}

is not a sequence, because it has the key 4 but does not have the key 3. (So, there is no n such that the set {1..n} is equal to the set of positive numeric keys of that table.) Note, however, that non-numeric keys do not interfere with whether a table is a sequence.
```

Which makes something like the implementation here (https://stackoverflow.com/a/52697380/558819) both shorter and faster (and you can pass in non-table-like objects).